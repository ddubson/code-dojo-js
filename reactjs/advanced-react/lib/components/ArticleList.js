import React from "react";
import { Article } from "./Article";

export const ArticleList = (props) => (
  <div>
    {Object.values(props.articles).map(article => {
      return <Article
        key={article.id}
        article={article}
        actions={props.articleActions}
      />
    })}
  </div>
);