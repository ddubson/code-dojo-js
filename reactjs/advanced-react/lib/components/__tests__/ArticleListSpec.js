import React from 'react';
import {shallow} from 'enzyme';
import {ArticleList} from '../ArticleList';


describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: {id: 'a', title: 'a', body: 'a', article: 'a'},
      b: {id: 'b', title: 'b', body: 'a', article: 'a'}
    },
    store: {
      lookupAuthor: jest.fn(() => ({}))
    }
  };

  it('renders correctly', () => {
    const wrapper = shallow(<ArticleList
      {...testProps}
    />);

    expect(wrapper.props().children.length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});