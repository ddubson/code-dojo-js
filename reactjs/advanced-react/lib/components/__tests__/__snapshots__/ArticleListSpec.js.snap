// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "a": Object {
          "article": "a",
          "body": "a",
          "id": "a",
          "title": "a",
        },
        "b": Object {
          "article": "a",
          "body": "a",
          "id": "b",
          "title": "b",
        },
      }
    }
    store={
      Object {
        "lookupAuthor": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Article
          article={
            Object {
              "article": "a",
              "body": "a",
              "id": "a",
              "title": "a",
            }
          }
        />,
        <Article
          article={
            Object {
              "article": "a",
              "body": "a",
              "id": "b",
              "title": "b",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "article": "a",
            "body": "a",
            "id": "a",
            "title": "a",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "article": "a",
            "body": "a",
            "id": "b",
            "title": "b",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Article
            article={
              Object {
                "article": "a",
                "body": "a",
                "id": "a",
                "title": "a",
              }
            }
          />,
          <Article
            article={
              Object {
                "article": "a",
                "body": "a",
                "id": "b",
                "title": "b",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "article": "a",
              "body": "a",
              "id": "a",
              "title": "a",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "article": "a",
              "body": "a",
              "id": "b",
              "title": "b",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
